### DISCUSSION TITLE
Qual o papel de um Tech Lead?

### DISCUSSION THREAD

**Discussion by gmoura** [frontendbr/forum#802] [Discussão]
Opa pessoal! 

Estive pensando sobre carreira profissional e sobre direcionar a carreira para ser um Tech Lead, e tenho duvidas sobre o assunto, tais como:

* Quais as responsabilidades?
* Quais os skills?

Se alguém puder compartilhar alguma experiência sobre como foi atuar como Tech Lead ou como foi trabalhar com um Tech lead, seria massa ;) 

Valeu, xablau

**Comment by felipefialho** [comment]
@danilodevhub

**Comment by matheusml** [comment]
Oi @gmoura 

A sua pergunta é difícil de ser respondida porque as responsabilidades e as _skills_ de um Tech Lead variam muito de empresa pra empresa.

Dito isso, vou contar como é a **minha experiência** em ser Tech Lead (de JavaScript) aqui na [Concrete](https://concrete.com.br/).

#### Responsabilidades
As responsabilidades que eu tenho hoje são:
- **Recrutamento**. Tenho que entrevistar possíveis candidatos, corrigir (e desenvolver) os desafios, etc
- **Guia**. Tenho que guiar tecnicamente outros colaboradores. Por exemplo, alguém está com uma dúvida se Angular ou React vai ser a melhor solução para um determinado projeto; como desenvolver testes unitários para componentes; como fazer _code-splitting_; etc.
- **Evangelização**. Tenho que ir em eventos, criar e manter projetos _open source_, criar conteúdo no _blog_ da Concrete. E demais atividades que chamem atenção para a empresa.

#### Skills
As _skills_ estão descritas [nesse documento](https://github.com/concretesolutions/front-end-guide) _open source_ que a gente criou recentemente. Mas só pra ter uma ideia de algumas das _skills_:
* Domina os Paradigmas de Programação e boas práticas com JavaScript
* Entende os problemas e limitações do JavaScript, CSS e HTML
* Consegue criar aplicações 100% com JavaScript e deployar
* Consegue conversar de igual para igual com um Dev Back-End
* Domina os Padrões de Projetos

Espero ter ajudado :+1:

**Comment by ghost** [comment]
Eu li algo uma vez que concordo totalmente:

> A Tech Lead is by definition, someone who is too good to be "just" a programmer, but not good enough to be more than a programmer. Once, you get there, you are parked in this halfway house of being part manager who isn't management, and part architect who isn't the architect. You stay there, until you build your skills far enough to move to what you are really good at.

Leitura complementar nessa thread: https://www.quora.com/How-do-I-become-a-great-tech-lead

**Responsabilidades:**

Coach e consultoria, você tem que tá disposto a lidar com outros programadores e muito das vezes entender que nem sempre o que você entende como melhor, realmente é. Liderança, afinal o nome já diz então você vai ter que correr risco e tem que tá preparado pra isso. Tenha empatia. Tente entender as motivações, distrações, objetivos e medos de outros.  Considero que isso já é metade do trabalho.

**Skills**

**Interpessoal** - Sim você vai lidar com pessoas e isso é muito importante pra você entender as forças do time e fraquezas.

**Técnico** - Esse não preciso nem falar, mas considero que mais do que ser bom tecnicamente é saber desde o inicio que você não será o melhor e muito menos que a sua voz deve ser a de um juiz e ser a última. Aqui entra o lado interpessoal será importante pra evoluir também o lado técnico.

**Dizer não** - Isso já é comum na vida de um programador, mas na vida de techlead é mais complicado. Já tive que dizer não ao dono de uma empresa e convencê-lo que não era por ser dono da empresa que o que ele tava pedindo tinha viabilidade técnica ou era o momento ou não pra fazer.
Isso acontece com dizer não a PO`s, PM`s, Diretores e etc.

Bom que eu lembre é isso, deve ter mais coisas com certeza, mas isso já ajuda dando minha opinião. @gmoura hehe

E o principio básico é:

> Líder tecnológico: oferece projetos liderando outros desenvolvedores de uma perspectiva técnica, incluindo seleção de tecnologia, estabelecimento de padrões de boa programação, estimativa realista e assistência contínua para resolução de problemas.

Ah e existem diferentes perfis de tech lead, talvez você terá que desenvolver o seu =)

**Comment by ghost** [comment]
@eduardojmatos talvez também possa contribuir.

**Comment by danilodevhub** [comment]
Olá @gmoura bacana seu interesse em direcionar a carreira para este lado. 

Concordo com o @matheusml que varia de empresa pra empresa. 
Complementando acho válido separar as palavras pra ficar didático:

- Tech: Um profissional com background técnico, que já tenha uma certa experiência com desenvolvimento e ter passado por diversas situações na engenharia de software. 
> Gosto de frisar esta questão técnica, porque já vi em empresas por aí alocarem líderes técnicos que não saibam escrever se quer uma linha de código ou que sabiam e deixaram esta disciplina de lado 

- Lead: Um profissional que seja e comporte-se como líder, ou seja, alguém que tenha poder de persuasão, atue de maneira consultiva, entenda e saiba direcionar os problemas e talvez o mais importante ser reconhecido como líder pelo próprio time.
> Aqui talvez é onde aparece o maior GAP dos profissionais de origem técnica e aí não tem jeito é desenvolver o tal [Soft Skill](https://en.wikipedia.org/wiki/Soft_skills)

Aqui no [Cubo](https://cubo.network/) as Responsabilidades são:

- Inicialmente montar o time do zero: identificar os perfis necessários, recrutar pessoas, etc;
- Definir arquitetura, tecnologias, padrões e boas práticas (aqui funcionamos de maneira muito colaborativa e temos um time bastante Senior, então não só eu, mas todos são responsáveis por esta parte);
- Ser consultivo também ao time de PM's levando um viés mais técnico aos produtos;
- E é claro "sujar a mão de graxa" como costumo dizer, nunca parar de codar 😃 

Recentemente saiu um podcast bacana sobre [O que faz um CTO?](https://hipsters.tech/o-que-faz-um-cto-hipsters-68/), que talvez seja um proximo passo na carreira de um Tech Lead, mas muitas das atribuições são comuns na minha visão.

**Comment by Cleysonlb** [comment]
@matheusml muito interessante esse link sobre os níveis de front end!!!

**Comment by eduardojmatos** [comment]
@edmolima obrigado pela lembrança ❤️ 

@gmoura na minha visão, de Tech Lead (Júnior ainda heheh), acho que as responsabilidades se resumem em:

- **Cuidar da saúde e felicidade do time**. Isso inclui proteger o time de possíveis histórias que comprometam a felicidade de se trabalhar na empresa, demandas top-down, entre outras coisas - até umas paradas de RH que estão atrapalhando a pessoa. Avaliar constantemente se todos estão felizes é uma das coisas que eu vejo de mais importante nesse perfil;

- **Ter a visão de tecnologia além das demandas diárias**. Isso significa que, quando uma história chega para ser avaliada pelo time, o Tech Lead precisa ter a visão de como isso vai escalar, qual arquitetura vai ser ideal para futuras alterações e quanto podemos implementar de ferramentas e tecnologias novas que resolvam a história de modo fácil - e que dure mais do que algum curto período de tempo;

- Os skills, os *soft* basicamente, são de entender conceitos de cultura de times, principalmente estar por dentro da metodologia ágil, pois já nos dá uma visão muito boa de desenvolvimento de software ágil (ágil != rápido), saber resolver conflitos entre pessoas, ser moderador(a) em discussões de tecnologia, conversar com pessoal de produto e chegar num meio-termo em histórias às vezes não tão bacanas pro time, enfim... Se olhar bem, são habilidades mais de negociação, de moderador(a) de alguém que seja imparcial. São coisas difíceis de conseguir, e leva tempo (estou aprendendo muito ainda heheh).

- Skills de tech, acho que sim, são bem necessárias, até pra poder conseguir debater e encontrar a melhor solução para o que for proposto fazer.

O que o @matheusml comentou, também faz parte. Embora eu veja algumas coisas, como "Evangelização", mais de um papel de um evangelista da empresa focado nisso. Recrutamento eu acho que é importante sim, mas envolver mais gente do time nesse processo faz mais sentido, na minha visão.

O @edmolima citou, de dizer não, acho que é primordial!

O @danilodevhub praticamente disse muita coisa que concordo. Inclusive de codar, afinal, o Tech Lead ainda precisa codar no time!

Enfim, espero ter colaborado!

**Comment by gmoura** [comment]
Massa galera, valeu por compartilhar um pouco das experiências de vocês <3

**Comment by rperucchi** [comment]
Boa Galera.
Adorei o tema. Recentemente me foi solicitado montar um workshop para techlead's e pesquisando acabei caindo aqui, ser techlead tende a variar muito de empresa para empresa e mesmo trabalhando com pessoas em outros países não vejo mudança no comportamento, ou seja, as pessoas tem que desenvolver o equilíbrio entre a liderança balanceado com conhecimento técnico. 
Vou me apoiar nestes pilares:
* Liderança 
* Skills
* Responsabilidades

Se tiverem alguma dica estou aberto.

**Comment by yuritoledo** [comment]
Mentoria de alguma stack (front ou back) para os mais novatos;
Cuidar dos empecilhos que acabam aparecendo para a equipe;
E os clássicos: Valorizar e motivar a equipe.

**Comment by wmartins** [comment]
Só contribuindo com a thread, o Pat Kua tem vários artigos e vídeos sobre isso, acho que é um bom caminho para quem quer entender melhor esse papel de Tech Lead. Vou deixar alguns links:

- https://www.thekua.com/atwork/2014/11/the-definition-of-a-tech-lead/

- https://www.thekua.com/atwork/2014/10/do-we-need-a-tech-lead/
- http://vvgomes.com/we-dont-need-tech-leads/ (réplica)
- https://www.thekua.com/atwork/2016/12/you-may-not-need-a-tech-lead-but-others-do/ (tréplica)

**Comment by rafaelmaruta** [comment]
olha só o @gmoura !

**Comment by georgebezerra** [comment]
Recentemente fui convidado para ser Tech Lead em um projeto. Aceitei sem saber nada de quais são as responsabilidades deste profissional, pois estou acostumado a trabalhar como desenvolvedor.  Pesquisando no Google achei esta issue que abriu a minha mente e pretendo colocar em prática todas as dicas aqui citadas. Muito..muito..muito "thanks"!!!!!

**Comment by gmoura** [comment]
Somando aqui o post https://medium.com/@gmoura/de-repente-tech-lead-1463653b8fe9
